pkg/controllers/vdb/init_db.go

```diff
@@ -269,12 +279,55 @@ func (g *GenericDatabaseInitializer) setKerberosAuthParms() {
 	g.ConfigurationParams.Set("KerberosEnableKeytabPermissionCheck", "0")
 }
 
+func (g *GenericDatabaseInitializer) setAuthFromGCSSecret(ctx context.Context) (string, ctrl.Result, error) {
+	name := "projects/otl-ot2-int-sb/secrets/new_vertica_secret/versions/1"
+	client, err := gsm.NewClient(ctx)
+	if err != nil {
+		return "", ctrl.Result{}, err
+	}
+	defer client.Close()
+
+	req := &secretmanagerpb.AccessSecretVersionRequest{
+		Name: name,
+	}
+
+	result, err := client.AccessSecretVersion(ctx, req)
+	if err != nil {
+		g.Log.Info("google: could not find default credentials")
+		return "", ctrl.Result{}, err
+	}
+
+	crc32c := crc32.MakeTable(crc32.Castagnoli)
+	checksum := int64(crc32.Checksum(result.Payload.Data, crc32c))
+	if checksum != *result.Payload.DataCrc32C {
+		return "", ctrl.Result{}, err
```

Comment:
err would be nil here. We should generate one:
```suggestion
		return "", ctrl.Result{}, fmt.Errorf("checksum failure, expected %d but got %d", *result.Payload.DataCrc32C, checksum)```

Reply:
Added err as suggested

Reply:
err is still nil here I believe.

Reply:
I know we talked about this on teams, but the code here looks wrong. I think my suggestion will solve it.

Reply:
Was a copy paster error, corrected it now